apps/emqx_modules/src/emqx_event_message.erl

```diff
@@ -172,7 +172,8 @@ on_message_acked(_ClientInfo = #{
     case ignore_sys_message(Message) of
         true -> ok;
         false ->
-            Payload0 = base_message(Message),
+            Message1 = remove_internal_headers(Message),
+            Payload0 = base_message(Message1),
```

Comment:
Maybe a better place to do the removal is in `base_message`?

Reply:
yes，i agree with you